docs/machine-learning/how-to-guides/consuming-model-ml-net.md

```diff
@@ -1,32 +1,160 @@
+mlContext.Model.Save(trainedModel, data.Schema, "model.zip");
+```
+
+Because most models and data preparation pipelines inherit from the same set of classes, the save and load operations for these components is the same. Depending on your use case, you can either combine the data preparation pipeline and model into a single [`EstimatorChain`](xref:Microsoft.ML.Data.TransformerChain%601) which would output a single [`ITransformer`](xref:Microsoft.ML.ITransformer) or separate them thus creating separate `ITransformers` for each. 
+
+## Save Model Locally
+
+When saving a model you need two things:
+
+1. The [`ITransformer`](xref:Microsoft.ML.ITransformer) of the model.
+2. The [`DataViewSchema`](xref:Microsoft.ML.DataViewSchema) of the [`ITransformer`](xref:Microsoft.ML.ITransformer)'s expected input.
+
+After training the model, use the [`Save`](xref:Microsoft.ML.ModelOperationsCatalog.Save*) method to save the trained model to a file called `model.zip` using the `DataViewSchema` of the data after the data preparation pipeline has been applied. 
+
+```csharp
+// Save Trained Model
+mlContext.Model.Save(trainedModel, data.Schema, "model.zip");
+```
+
+## Load Model Stored Locally
+
+Models stored locally can be used in other processes or applications like `ASP.NET Core` and `Serverless Web Applications`. See [Use ML.NET in Web API](./serve-model-web-api-ml-net.md) and [Deploy ML.NET Serverless Web App](./serve-model-serverless-azure-functions-ml-net.md) how-to articles to learn more. 
+
+In a separate application or process, use the [`Load`](xref:Microsoft.ML.ModelOperationsCatalog.Load*) method along with the file path to get the trained model into your application.
+
+```csharp
+//Define DataViewSchema for data preparation pipeline and trained model
+DataViewSchema modelSchema;
+
+// Load trained model
+ITransformer trainedModel = mlContext.Model.Load("model.zip", out modelSchema);
+```
+
+## Load Model Stored Remotely
+
+To load data preparation pipelines and models stored in a remote location into your application, use a [`Stream`](xref:System.IO.Stream) instead of a file path in the [`Load`](xref:Microsoft.ML.ModelOperationsCatalog.Load*) method.
 
 ```csharp
-using (var stream = File.Create(modelPath))
+// Create MLContext
+MLContext mlContext = new MLContext();
+
+// Define DataViewSchema and ITransformers
+DataViewSchema modelSchema;
+ITransformer trainedModel;
+
+// Load data prep pipeline and trained model 
+using (HttpClient client = new HttpClient())
 {
-    // Saving and loading happens to 'dynamic' models.
-    mlContext.Model.Save(model, stream);
+    Stream modelFile = await client.GetStreamAsync("<YOUR-REMOTE-FILE-LOCATION>");
+
+    trainedModel = mlContext.Model.Load(modelFile, out modelSchema);
 }
+```
+
+## Separate Data Preparation Pipelines and Models
+
+When working with separate data preparation pipelines and models, the same operations apply, except they both now components need to be saved and loaded simultaneously.
```

Comment:
>they both now components need [](start = 100, length = 29)

this gives me the impression that the components of the pipelines need to be separately saved and loaded. 

Maybe rephrase to make it more clear that it is the pipelines that need to be saved/loaded.

Reply:
#resolved